packages/pymedphys_mudensity/src/pymedphys_mudensity/plt/helpers.py

```diff
@@ -29,8 +29,10 @@
 
 def pcolormesh_grid(x, y, grid_resolution=None):
     if grid_resolution is None:
-        grid_resolution = x[1] - x[0]
-        assert grid_resolution == y[1] - y[0]
+        diffs = np.hstack([np.diff(x), np.diff(y)])
+        assert np.all(np.abs(diffs - diffs[0]) < 10**-12)
```

Comment:
Not at all required, but [`numpy.allclose`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.allclose.html) is perhaps neater here?

Reply:
I thought about that, but it's actually not as neat. All close I believe requires two arrays to be the same size. This is an array and a scalar. I could be wrong though.

Reply:
Numpy accepts a scalar in either position. It simply broadcasts.

I think that's the point of the "array or *array-like*" phrase present in many numpy parameter docstring sections - so long as it can broadcast, it's fine. Scalars should generally work, I think.

https://stackoverflow.com/questions/41363751/numpy-is-there-an-allclosenp-array-scalar

Reply:
Thanks for that @Matthew-Jennings I believe you're right :).